src/Core/Checkout/Cart/CartSerializationCleaner.php

```diff
@@ -88,4 +92,15 @@ private function cleanupCustomFields(LineItem $lineItem, array $customFieldAllow
             )
         );
     }
+
+    private function cleanupCartError(\Exception $error): void
+    {
+        $traces = $error->getTrace();
+        foreach ($traces as &$trace) {
+            $trace['args'] = [];
+        }
+
+        $traceProperty = new \ReflectionProperty(\Exception::class, 'trace');
+        $traceProperty->setValue($error, $traces);
+    }
```

Comment:
I find this really ugly, I have to say. I think it is prettier, if we implement a `__serialize()` magic method into the Cart error. Something like this:
```php
public function __serialize() {
    return get_object_vars($this);
}
```

I would also be open to add a deprecation for 6.8 and to remove the inheritance of cart-`Error` from `Exception` and move to a `Struct`. I think, CartErrors are not thrown anywhere, so it should be fine.